include/deal.II/base/function_parser.h

```diff
@@ -26,10 +26,7 @@
 #include <vector>
 #include <map>
 
-namespace mu
-{
-  class Parser;
-}
+#include <muParser.h>
```

Comment:
There is a difference between instantiating the *type* and individual *functions* or other members. Clearly, when you compile a member function that calls `std::vector::back()`, you will need to know. But these functions could all reside in the `.cc` file where we do include `muParser.h`. If these functions are in the `.h` file, then that seems like the wrong place.